Userland/Libraries/LibCore/Socket.cpp

```diff
@@ -370,6 +370,9 @@ ErrorOr<pid_t> LocalSocket::peer_pid() const
 #elif defined(AK_OS_NETBSD)
     struct sockcred creds = {};
     socklen_t creds_size = sizeof(creds);
+#elif defined(AK_OS_SOLARIS)
+    ucred_t* creds = NULL;
```

Comment:
This seems wrong.  Man page for solaris does not mention this option (https://docs.oracle.com/cd/E86824_01/html/E54774/getsockopt-3socket.html ) I did found that solaris (!?) does not support credentials:

https://vistech.net/~champ/online-docs/books/SecProgCnC++/0596003943_secureprgckbk-chp-9-sect-8.html

If modern indiana does support... please point me to its document. If `SCM_CREDS` is not supported - just return an error from this API (this is what I was thinking of doing in the win32 port).

Reply:
It does support credentials,see here https://docs.oracle.com/cd/E88353_01/html/E37843/getsockopt-3c.html the last point,it's called SO_RECVUCRED.
That one did *really* take a long time to research and make it work,but I actually run OpenIndiana on my primary computer and verified every change I did.